classes/controlledVocab/ControlledVocab.php

```diff
@@ -1,108 +1,201 @@
+        ];
     }
 
     /**
-     * Set associated type.
-     *
-     * @param int $assocType
+     * Check if a provided vocab symbolic defined in pre defined symbolic list
      */
-    public function setAssocType($assocType)
+    public static function hasDefinedVocabSymbolic(string $vocab): bool
     {
-        $this->setData('assocType', $assocType);
+        return in_array($vocab, static::getDefinedVocabSymbolic());
     }
 
     /**
-     * Get symbolic name.
-     *
-     * @return string
+     * Get all controlled vocab entries for this controlled vocab
      */
-    public function getSymbolic()
+    public function controlledVocabEntries(): HasMany
     {
-        return $this->getData('symbolic');
+        return $this->hasMany(ControlledVocabEntry::class, 'controlled_vocab_id', 'controlled_vocab_id');
     }
 
     /**
-     * Set symbolic name.
-     *
-     * @param string $symbolic
+     * Scope a query to only include vocabs with a specific symbolic.
+     */
+    public function scopeWithSymbolic(Builder $query, string $symbolic): Builder
+    {
+        return $query->where('symbolic', $symbolic);
+    }
+
+    /**
+     * Scope a query to only include vocabs with a specific assoc type and assoc ID.
      */
-    public function setSymbolic($symbolic)
+    public function scopeWithAssoc(Builder $query, int $assocType, int $assocId): Builder
     {
-        $this->setData('symbolic', $symbolic);
+        return $query
+            ->where('assoc_type', $assocType)
+            ->where('assoc_id', $assocId);
+    }
+
+    /**
+     * Scope a query to only include vocabs associated with given context id
+     */
+    public function scopeWithContextId(Builder $query, int $contextId): Builder
+    {
+        return $query
+            ->where(
+                fn ($query) => $query
+                    ->select('context_id')
+                    ->from('submissions')
+                    ->whereColumn(
+                        DB::raw(
+                            "(SELECT publications.submission_id 
+                            FROM publications 
+                            INNER JOIN {$this->table} 
+                            ON publications.publication_id = {$this->table}.assoc_id 
+                            LIMIT 1)"
+                        ),
+                        '=',
+                        'submissions.submission_id'
+                    ), 
+                $contextId
+            );
     }
 
     /**
      * Get a list of controlled vocabulary options.
      *
-     * @param string $settingName optional
-     *
      * @return array $controlledVocabEntryId => name
      */
-    public function enumerate($settingName = 'name')
-    {
-        $controlledVocabDao = DAORegistry::getDAO('ControlledVocabDAO'); /** @var ControlledVocabDAO $controlledVocabDao */
-        return $controlledVocabDao->enumerate($this->getId(), $settingName);
-    }
-}
+    public function enumerate(?string $settingName = null): array
```

Comment:
Here `$settingName` previously have default value to `name` but that seems not right as for controlled vocab, we have a specific sets to settings only . Or I am missing something ?